### PR TITLE
det-2342: async jobs and scm exceptions in sdk/cli/tests

### PR DESCRIPTION
**Pull request by makk94**
it's not all the new stuff, but it's some of it :)

### CODE REVIEW

File: python/sdk/prelude_sdk/controllers/scm_controller.py [hunk 1]
```diff
@@ -125,6 +125,102 @@ def update_evaluation(self, partner: Control):
...
+            json=body,
+            headers=self.account.headers,
+            timeout=10
+        )
+        if res.status_code == 200:
+            return res.json()
+        raise Exception(res.text)
+
+    @verify_credentials
+    def update_object_exception(self, exception_id, expires = '', filter = None, name = None):
```
Comment (tangtan2) [inline python/sdk/prelude_sdk/controllers/scm_controller.py:162]: why a string here but None in create?

File: python/cli/prelude_cli/views/scm.py [hunk 2]
```diff
@@ -107,11 +113,16 @@ def sync(controller, partner):
...
     """ Export SCM data """
     with Spinner(description='Exporting SCM data'):
+        detect = DetectController(account=controller.account)
         export = ExportController(account=controller.account)
-        data = export.export_scm(export_type=type, filter=odata_filter, orderby=odata_orderby, partner=Control[partner] if partner else None, top=limit)
-        with open(output_file, 'wb') as f:
-            f.write(data)
-        return dict(status=True), f'Exported data to {output_file}'
+        job_id = export.export_scm(export_type=type, filter=odata_filter, orderby=odata_orderby, partner=Control[partner] if partner else None, top=limit)['job_id']
+        while (result := detect.get_background_job(job_id))['end_time'] is None:
```
Comment (tangtan2) [inline python/cli/prelude_cli/views/scm.py:119]: is this going to spam get background job, should we put a small sleep here?

File: python/sdk/prelude_sdk/controllers/scm_controller.py (hunk 1 above)
Comment (makk94) [reply python/sdk/prelude_sdk/controllers/scm_controller.py:162]: it's a weird/bad workaround to knowing if the CLI is passing through an 'expires' param that is actively None (ie. don't expire ever) vs not passing through expires at all (ie. leave the value alone)

could also make a BE change to read "9999-12-12" (datetime.max) as "never expire" and then None can can be read as "leave the value alone"... both are kinda awkward for different reasons

File: python/cli/prelude_cli/views/scm.py (hunk 2 above)
Comment (makk94) [reply python/cli/prelude_cli/views/scm.py:119]: good idea. you are more polite than me

File: python/sdk/prelude_sdk/controllers/scm_controller.py (hunk 1 above)
Comment (tangtan2) [reply python/sdk/prelude_sdk/controllers/scm_controller.py:162]: hmmmmm i see, then should this also be changed to string in the put policy exception function? and the same check if expires == ''

Comment (makk94) [reply python/sdk/prelude_sdk/controllers/scm_controller.py:162]: the policy exception is a full on put, right? expires should always be set and I'll just remove the default altogether

Comment (tangtan2) [reply python/sdk/prelude_sdk/controllers/scm_controller.py:162]: yeah it's a full put

Comment (tangtan2) [reply python/sdk/prelude_sdk/controllers/scm_controller.py:162]: are you also going to do the 'arg' thing?

**Review by tangtan2 — APPROVED** [review]

**Pull request merged**